source/nodejs/adaptivecards-designer/src/card-designer.ts

```diff
@@ -449,9 +457,21 @@ export class CardDesigner extends Designer.DesignContext {
     }
 
     private targetVersionChanged() {
+        // Question: is there a specific reason that we get the payload from the designerSurface first?
```

Comment:
Does anyone know why we do this?

Reply:
We're overriding the version in the payload (`cardPayload["version"] = ...`) to respond to the user changing the target version. Presumably this is so that we get accurate parsing behavior?

Reply:
I guess I just don't understand why we don't grab the payload from the editor directly.

I don't think it really matters outside of the array scenario though